.github/workflows/ci_cd.yml

```diff
@@ -0,0 +1,110 @@
+name: Client & CMS CI/CD
+
+on:
+  workflow_dispatch:
+  push:
+
+jobs:
+  deploy:
+    name: Deploy Client app
+    # if: (github.head_ref || github.ref_name) == 'main' || (github.head_ref || github.ref_name) == 'staging'
+    # if: (github.head_ref || github.ref_name) == 'staging'
+    if: (github.head_ref || github.ref_name) == 'infrastructure'
+    runs-on: ubuntu-latest
```

Comment:
you-in-6-months will have a great time with this

Reply:
Yeah, I am always split between using latest stuff and discover some potential issue or hardcode some version and then be surprised when it gets super difficult to upgrade. Don't mind to hardcode the latest versions here...

Reply:
I get your point, but in this particular case using "latest" might cause CD to not work for no good reason. For example, you could have tests run and fail on future ubuntu, but the deployment would still work because our Docker images use pinned OS versions (sort of).

Reply:
So you would propose to use same OS which dockers are using? Makes sense to me...

Reply:
my previous comment is a bit misleading - it's not only that the tests may fail, but also that the actual steps of the GH action may fail as a result of the invisible OS change - say, for example, your custom filter action, for some reason, does not work on ubuntu 24.04 whenever that comes out.

Reply:
Yes, that could happen and it goes to my first comment -- either you discover this issue when new version of OS comes out, it stops working and theoretically hardcode version here afterwards if it gets too difficult to fix it immediately or hardcode version here now and be oblivious about what works with which version. In second case, when you will be forced to upgrade, you can theoretically have hard time. I honestly don't mind to use latest version or hardcode version here, always not sure which option is better for my future self :D

Reply:
I see your point, and on one side I do like the "move fast and break things" approach, but then we should also throw yarn.lock out the window ;) The reason we use version locks it is not because we don't want to upgrade our code, or not because we don't want to know about breaking changes ASAP - it is so we can tackle those things when it's convenient for us.

pin it, please

Reply:
😄 will do, no problem, thanks for your feedback on this